x-pack/plugins/apm/server/lib/helpers/get_document_sources.ts

```diff
@@ -111,21 +175,33 @@ export async function getDocumentSources({
       rollupInterval,
       hasDataBefore,
       hasDataAfter,
+      checkSummaryExistsField: source.meta.checkSummaryExistsField,
     };
   });
 
-  const hasAnyDataBefore = checkedSources.some(
-    (source) => source.hasDataBefore
-  );
+  const hasAnyDataBefore = checkedSources
+    .filter((source) => !source.checkSummaryExistsField)
+    .some((source) => source.hasDataBefore);
 
-  const sources: Array<ApmDataSource & { hasDocs: boolean }> =
-    checkedSources.map((source) => {
-      const { documentType, hasDataAfter, hasDataBefore, rollupInterval } =
-        source;
+  const hasAnySummaryDataBefore = checkedSources
```

Comment:
@dgieselaar  I still had to split this in 2 different checks due to the comment present in line 207-212 in this file. As this filter checks for docs to be present in and since we are merging both set of API calls for summary field and docs field, i am not 100% sure if this single filter would be correct